lib/ansible/modules/dnf.py

```diff
@@ -531,6 +528,21 @@ def _compare_evr(self, e1, v1, r1, e2, v2, r2):
         return rc
 
     def _ensure_dnf(self):
+        locale = get_best_parsable_locale(self.module)
+        os.environ['LC_ALL'] = os.environ['LC_MESSAGES'] = os.environ['LANG'] = locale
+
+        global dnf
+        try:
+            import dnf
+            import dnf.cli
+            import dnf.const
+            import dnf.exceptions
+            import dnf.subject
+            import dnf.util
+            HAS_DNF = True
+        except ImportError:
+            HAS_DNF = False
```

Comment:
I think this should be more distinct:
```suggestion
        except ImportError:
            HAS_DNF = False
        else:
            HAS_DNF = True
```

Reply:
I don't mind the current structure. I'd even be ok with setting `HAS_DNF = True` outside the `try` `except`. `¯\_(ツ)_/¯`

Reply:
I just copied this code over :-)